src/Hosting/Application.php

```diff
@@ -305,4 +306,25 @@ public function setRealm(RealmInterface $realm)
     {
         $this->realm = $realm;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getEnvironmentList()
+    {
+        if ($this->environmentList === null) {
+            throw new \RuntimeException(
+                sprintf('%s: This Application object does not know the environment list.', __METHOD__)
+            );
+        }
+        return $this->environmentList;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setEnvironmentList(Environment\EnvironmentListInterface $environmentList)
```

Comment:
can you please add `use Acquia\Platform\Cloud\Hosting\Environment\EnvironmentListInterface` at top and just use `EnvironmentListInterface` here? (Interface also should be tweaked, but this looks correct in decorator)